src/Frontend/Page/Topic.hs

```diff
@@ -279,9 +264,12 @@ viewTopic tab topicId = do
                 pure $ ViewTopicDelegations topic delegations
             _ ->
               do
-                let q = fromMaybe (Nothing, Nothing) $ tab ^? viewTopicTabQuery
-                ideas <- ideasRunQuery q <$> findIdeasByTopic topic
-                ideasAndNumVoters <- ListItemIdeas ctx q <$> (getListInfoForIdea `mapM` ideas)
+                let loc = topicIdeaLocation topic
+                    ideasQuery = fromMaybe (assert False $ error "viewTopic: impossible.")
+                               $ tab ^? viewTopicTabQuery
```

Comment:
> You can use: `tab ^?! viewTopicTabQuery`

yes, but that gives the same bad error messages as fromJust.  i'd rather like to keep this noise.  (unless you write enough tests to convince me we'll never run into that problem ;-)